cmd/syft/cli/commands/packages.go

```diff
@@ -205,51 +207,228 @@ func getSource(opts *options.Catalog, userInput string, filters ...func(*source.
 }
 
 func generateSBOM(id clio.Identification, src source.Source, opts *options.Catalog) (*sbom.SBOM, error) {
-	tasks, err := eventloop.Tasks(opts)
+	s, err := syft.CreateSBOM(src, opts.ToSBOMConfig(id))
 	if err != nil {
+		expErrs, err := filterExpressionErrors(err, 0)
+		notifyExpressionErrors(expErrs)
```

Comment:
The current output looks something like this (the section "Found 1 invalid cataloger selection expression" is purple while the error section starting with "1 error occurred:" is red):
```
$ syft alpine:latest --override-default-catalogers js
...
Found 1 invalid cataloger selection expression:            
                                                           
 ❖ Given expression "--override-default-catalogers js"     
   However, "js" is not a recognized cataloger name or tag.
                                                           
1 error occurred:
        * invalid cataloger selection expression provided
```
I think it _may_ be more clear if the errors remain just as they are and we supplement the messaging to the user with suggestions in the purple section. I'd suggest something along these lines (the purple section is the Hints or Suggestions, and the red is the 1 error occurred, like it is today:

```
$ syft alpine:latest --override-default-catalogers js
...
1 error occurred:
        * "js" is not a recognized cataloger name or tag. default catalogers must be specified by name or tag
```

```
$ syft alpine:latest --select-catalogers sbom-cataloger
...
Hints:
        - to add the sbom-cataloger to the defaults use: --select-catalogers +sbom-cataloger

1 error occurred:
        * "sbom-cataloger" is not a known cataloger tag. filtering catalogers must use tags
```

```
$ syft alpine:latest --select-catalogers +javascript
...
Suggestions:
        - to add all javascript catalogers to the defaults use: --override-default-catalogers default,javascript

1 error occurred:
        * "javascript" is not a recognized cataloger name. adding catalogers must be specified by name
```

... as discussed, this would also mean all the code to remove/alter the errors can go away and just check for the presence of the known errors to populate the optional suggestion section.

Reply:
I've updated the code with the following changes:
- original errors are persisted, which makes the code much easier to follow
- only suggestions that are actionable are shown, the original errors will enumerate any non-actionable issues

I didn't rework the copy to be exactly what you had above. Here's an example:


```
❯ syft alpine:3.12 --override-default-catalogers image --select-catalogers javascript --select-catalogers python,sbom-cataloger --select-catalogers +js,+javascript
...

Suggestions:                                                                                              
                                                                                                          
 ❖ Given expression "--select-catalogers sbom-cataloger"                                                  
   However, names are not allowed with this operation (must use tags).                                    
   It seems like you are intending to add a cataloger in addition to the default set.                     
   ... Did you mean "--select-catalogers +sbom-cataloger" instead?                                        
                                                                                                          
 ❖ Given expression "--select-catalogers javascript"                                                      
   However, tags are not allowed with this operation (must use exact names).                              
   Adding groups of catalogers may result in surprising behavior (create inaccurate SBOMs).               
   If you are certain this is what you want to do, use "--override-default-catalogers javascript" instead.
                                                                                                          
                                                                                                          
3 errors occurred:
        * invalid expression: "sbom-cataloger": names are not allowed with this operation (must use tags)
        * invalid expression: "javascript": tags are not allowed with this operation (must use exact names)
        * invalid expression: "js": unknown name or tag given

```